### PR TITLE
Heat-208: updated alert rules

### PR DESCRIPTION
**Pull request by KalyanaVadlamani**

### CODE REVIEW

File: charts/generic-prometheus-alerts/templates/ingress-alerts.yaml [hunk 1]
```diff
@@ -28,6 +28,17 @@ spec:
           for: 1m
           labels:
             severity: {{ .Values.alertSeverity }}
+        - alert: High503ErrorRateOnHealthEndpoint
+          annotations:
+            message: High rate of 503 errors detected on /health path in Ingress {{`{{`}} $labels.exported_namespace {{`}}`}}/{{`{{`}} $labels.ingress {{`}}`}}.
+            runbook_url: {{ .Values.runbookUrl }}ingress-503-error-responses
+            dashboard_url: {{ $.Values.grafanaUrl }}/d/golden-signals/golden-signals?orgId=1&var-namespace={{ $targetNamespace }}&var-service={{ $targetApplication }}
+          expr: |-
+            avg(rate(nginx_ingress_controller_requests{exported_namespace=~"{{ .Values.nonProdNamespacePattern }}", ingress=~"{{ $targetIngress }}", path=~"/health", status=~"503"}[{{ .Values.ingress503ErrorWindowMinutes }}m]) > 0.1) by (ingress, exported_namespace)
```
Comment (mattops) [inline charts/generic-prometheus-alerts/templates/ingress-alerts.yaml:37]: path="/health"

Comment (KalyanaVadlamani) [reply charts/generic-prometheus-alerts/templates/ingress-alerts.yaml]: updated. 
nonProdNamespacePattern --> if removed, we will have 2 rules
rule 1: for 5xx errors
rule 2: for 503 errors, leading to increasing the noise

File: charts/generic-prometheus-alerts/templates/ingress-alerts.yaml [hunk 2]
```diff
@@ -28,6 +28,17 @@ spec:
           for: 1m
           labels:
             severity: {{ .Values.alertSeverity }}
+        - alert: High503ErrorRateOnHealthEndpoint
+          annotations:
+            message: High rate of 503 errors detected on /health path in Ingress {{`{{`}} $labels.exported_namespace {{`}}`}}/{{`{{`}} $labels.ingress {{`}}`}}.
+            runbook_url: {{ .Values.runbookUrl }}ingress-503-error-responses
+            dashboard_url: {{ $.Values.grafanaUrl }}/d/golden-signals/golden-signals?orgId=1&var-namespace={{ $targetNamespace }}&var-service={{ $targetApplication }}
+          expr: |-
+            avg(rate(nginx_ingress_controller_requests{exported_namespace=~"{{ .Values.nonProdNamespacePattern }}", ingress=~"{{ $targetIngress }}", path=~"/health", status="503"}[{{ .Values.ingress503ErrorWindowMinutes }}m]) > 0.1) by (ingress, exported_namespace)
```
Comment (mattops) [inline charts/generic-prometheus-alerts/templates/ingress-alerts.yaml:37]: If you set `path="/health"` here - and then `path="/"` on the other rule. - it should separate them nicely.

Comment (mattops) [reply charts/generic-prometheus-alerts/templates/ingress-alerts.yaml:37]: then we can have `status=~"5.*"` on both rules.

File: charts/generic-service/values.yaml
Comment (KalyanaVadlamani) [inline charts/generic-service/values.yaml:1]: /info introduced for future use

File: charts/generic-service/values.yaml [hunk 3]
```diff
@@ -59,6 +59,8 @@ ingress:
   host: chart-example.local
   hosts: []
   path: /
+  health_path: /health
+  info_path: /info
```
Comment (babuharidass11) [inline charts/generic-service/values.yaml:63]: can we use camelcase variable names here?

File: charts/generic-prometheus-alerts/values.yaml [hunk 4]
```diff
@@ -120,5 +120,14 @@ rdsAlertsConnectionThresholdMinutes: 5
 # Set the error window minutes for 5xx ingress
 ingress5xxErrorWindowMinutes: 1
 
+# Set the error window minutes for 503 ingress High503ErrorRateOnHealthEndpoint
+ingress503ErrorWindowMinutes: 5
+
+# Set the error
+# nonProdNamespacePattern: ".*-(dev|preprod)$"
```
Comment (babuharidass11) [inline charts/generic-prometheus-alerts/values.yaml:127]: are these line still required?

File: charts/generic-prometheus-alerts/templates/ingress-alerts.yaml [hunk 5]
```diff
@@ -23,11 +23,23 @@ spec:
...
             dashboard_url: {{ $.Values.grafanaUrl }}/d/golden-signals/golden-signals?orgId=1&var-namespace={{ $targetNamespace }}&var-service={{ $targetApplication }}
           expr: |-
-            avg(rate(nginx_ingress_controller_requests{exported_namespace=~"{{ $targetNamespace }}", ingress=~"{{ $targetIngress }}", status=~"5.*"}[{{ .Values.ingress5xxErrorWindowMinutes }}m]) > 0) by (ingress, exported_namespace)
+            avg(rate(nginx_ingress_controller_requests{exported_namespace=~"{{ $targetNamespace }}", ingress=~"{{ $targetIngress }}", path="/", status=~"5.*"}[{{ .Values.ingress5xxErrorWindowMinutes }}m]) > 0) by (ingress, exported_namespace)
             {{ $businessOrAllHoursExpression }}
           for: 1m
           labels:
             severity: {{ .Values.alertSeverity }}
+            
+        - alert: 503ErrorResponsesOnHealthEndpoint
```
Comment (mattops) [inline charts/generic-prometheus-alerts/templates/ingress-alerts.yaml:32]: Should this name be 5xxError...?

File: charts/generic-prometheus-alerts/templates/ingress-alerts.yaml [hunk 6]
```diff
@@ -23,11 +23,23 @@ spec:
...
+          annotations:
+            message: High rate of 503 errors detected on /health path in Ingress {{`{{`}} $labels.exported_namespace {{`}}`}}/{{`{{`}} $labels.ingress {{`}}`}}.
```
Comment (mattops) [inline charts/generic-prometheus-alerts/templates/ingress-alerts.yaml:34]: Same here 503?

File: charts/generic-prometheus-alerts/templates/ingress-alerts.yaml [hunk 7]
```diff
@@ -23,11 +23,23 @@ spec:
...
+            runbook_url: {{ .Values.runbookUrl }}ingress-503-error-responses
+            dashboard_url: {{ $.Values.grafanaUrl }}/d/golden-signals/golden-signals?orgId=1&var-namespace={{ $targetNamespace }}&var-service={{ $targetApplication }}
```
Comment (mattops) [inline charts/generic-prometheus-alerts/templates/ingress-alerts.yaml:36]: Be good to link to this dashboard, and interpolate the namespace and ingress name so it loads with the correct option selected.

```
https://grafana.live.cloud-platform.service.justice.gov.uk/d/f1e13059dfd23fdcaf479f4fa833f92610c2dfa5/kubernetes-ingress-traffic?orgId=1&var-namespace=hmpps-adjustments-dev&var-ingress=hmpps-adjustments-v1-2
```

File: charts/generic-prometheus-alerts/values.yaml [hunk 8]
```diff
@@ -120,5 +120,8 @@ rdsAlertsConnectionThresholdMinutes: 5
 # Set the error window minutes for 5xx ingress
 ingress5xxErrorWindowMinutes: 1
 
+# Set the error window minutes for 503 ingress High503ErrorRateOnHealthEndpoint
+ingress503ErrorWindowMinutes: 5
```
Comment (mattops) [inline charts/generic-prometheus-alerts/values.yaml:124]: 503? maybe `ingressHealth5xxErrorWindowMinutes`

**Review by mattops — APPROVED** [review]

**Pull request merged**